### ISSUE TITLE
`escape-inline-tags` incorrectly reports `@` inside Markdown code spans

### ISSUE THREAD

**Issue by Pixel998** [gajus/eslint-plugin-jsdoc#1723]
## Expected behavior

The `escape-inline-tags` rule should ignore `@` characters that appear inside Markdown code spans. For example:

```js
/**
 * @file Main CLI that is run via the `npm init @eslint/config` command.
 */
```

## Actual behavior

The rule reports:

```
Unexpected inline JSDoc tag. Did you mean to use {@eslint}, \@eslint, or `@eslint`?
```

## ESLint Config

```js
import {jsdoc} from 'eslint-plugin-jsdoc';

export default [
  jsdoc({
    config: 'flat/recommended',
  })
];
```

## ESLint sample

```js
/**
 * @file Main CLI that is run via the `npm init @eslint/config` command.
 */
```

## Environment

- Node version: v24.16.0
- ESLint version v10.7.0
- `eslint-plugin-jsdoc` version: v63.0.13

**Comment by lumirlumir** [comment]
To add a bit more context to this issue, it comes from https://github.com/eslint/create-config/pull/255#discussion_r3566448490.
